waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -13,28 +13,30 @@ import qualified System.Directory
 import qualified System.FilePath as FP
 import Text.Printf (printf)
 import Wasp.Analyzer.Parser (isValidWaspIdentifier)
+import Wasp.Analyzer.Parser.Token (prettyShowKeyword)
 import Wasp.AppSpec.ExternalCode (SourceExternalCodeDir)
 import Wasp.Cli.Command (Command, CommandError (..))
 import qualified Wasp.Cli.Command.Common as Command.Common
 import qualified Wasp.Cli.Common as Common
 import qualified Wasp.Data
-import Wasp.Util (indent)
+import Wasp.Util (allValues, indent, quoteSingle)
 import qualified Wasp.Util.Terminal as Term
 
 newtype ProjectName = ProjectName {_projectName :: String}
 
 createNewProject :: String -> Command ()
 createNewProject name
   | not (isValidWaspIdentifier name) =
-      throwError $
-        CommandError "Project creation failed" $
-          intercalate
-            "\n"
-            [ "The project's name must be a valid Wasp identifier:",
-              indent 2 "- It can start with a letter or an underscore.",
-              indent 2 "- It can contain only letters, numbers, or underscores.",
-              indent 2 "- It can't be a Wasp keyword: 'import', 'from', 'true', 'false'."
-            ]
+      let keywordList = intercalate ", " $ map (quoteSingle . prettyShowKeyword) (allValues :: [Keyword])
```

Comment:
For aesthetics, I would do:

```suggestion
      let keywordList = intercalate ", " $ quoteSingle . prettyShowKeyword <$> (allValues :: [Keyword])
```

Reply:
I just love using `<$>` next to `$` because it basically says: apply thing on the left to the thing on the right, but through something. So they are very similar, one just "passes through". So if I see `a $ b <$> c $ d` I just read it like "apply c to d, then b to that, and then a to that" and I don't even think too much if it is functor or not, is just application happening.
So I guess that uniformity is nice, I like it!
I think ++ vs <> is a bit different, mappend on other stuff is less common I think. But now that you say it I am tempted to use <> more instead of ++ :D.
Btw I don't have a strong opinion here, this is just subjective preference, whatever you like better is good with me.

Reply:
Good points. I have a few counterpoints, but let's leave them for another time to keep things interesting (we're not merging this one anyway).

> But now that you say it I am tempted to use <> more instead of ++ :D.

Haha, this reminds me of when I was looking through some bioinformatics-related C++ code and stumbled upon `struct's with methods (as opposed to classes that I'd normally expect). Anyway, I was horrified and showed it to my friend, saying, "Look at what these crazy guys are doing. It doesn't make any sense." Then, one week later, I catch him writing `struct's with methods.

Reply:
Haha yes that depicts it nicely! I remember myself also not liking use of `fmap` when `map` can do the job, same for people using `<>`, but how the tables have turned!